vhdbuilder/packer/test/windows-vhd-content-test.ps1

```diff
@@ -210,7 +210,20 @@ function Test-ImagesPulled {
         $pulledImages = (ctr.exe -n k8s.io image ls -q | Select-String -notmatch "sha256:.*" | % { $_.Line } )
     }
     elseif ($containerRuntime -eq 'docker') {
-        $pulledImages = docker images --format "{{.Repository}}:{{.Tag}}"
+        $MaxRetryCount = 10
+        $RetryDelaySeconds = 10
+        $retryCount=0
+        do {
+            $pulledImages = docker images --format "{{.Repository}}:{{.Tag}}" 2> dockerError.log
+            $SEL = Select-String -Path dockerError.log -Pattern "docker daemon is not running"
+            if ($SEL -eq $null){
+                break
+            }
+            Write-Output "Failed to list docker images for docker daemon is not running"
```

Comment:
Could we check the status of the docker service and then pull the images?

Reply:
`list image` provides the status of docker implicitly, I think.